.eslintrc.js

```diff
@@ -92,11 +92,24 @@ module.exports = {
       }
     },
     {
-      files: ['**/govuk/components/**/*.{cjs,js,mjs}'],
-      excludedFiles: ['**/govuk/components/**/*.test.{cjs,js,mjs}'],
+      files: ['src/govuk/**/*.{cjs,js,mjs}'],
+      excludedFiles: ['src/govuk/**/*.test.{cjs,js,mjs}'],
       parserOptions: {
         // Note: Allow ES6 for import/export syntax (although our code is ES3 for legacy browsers)
         ecmaVersion: '2015'
+      },
+      plugins: ['es-x'],
+      extends: [
+        'plugin:es-x/restrict-to-es3'
+      ],
+      rules: {
+        // We use ES6 modules to structure our code
+        'es-x/no-modules': 'off',
+        // Rollup plays fine with `export * from`
+        'es-x/no-export-ns-from': 'off',
+        // These ES5+ features are polyfilled
+        'es-x/no-function-prototype-bind': 'off',
+        'es-x/no-date-now': 'off'
```

Comment:
@romaricpascal I'm aware we've got a much wider list of polyfills in our repo

>~Date.now()~ ✅
>Element.prototype.classList
>Element.prototype.closest()
>Element.prototype.dataset()
>Element.prototype.matches()
>Element.prototype.nextElementSibling
>Element.prototype.previousElementSibling
>~Function.prototype.bind()~ ✅
>Object.defineProperty()
>String.prototype.trim()

Plus `Document`, `DOMTokenList`, `Element`, `Event`, `Window` polyfills

I'm worried now that this plugin isn't catching those missed ones?

Reply:
Yeah, I'm not seeing it as an addition that'll catch everything, just automate some checks so we can focus on what actually requires a human eye. There's the caveat I added back in the description that the plugin only having rules for language features, not browser API features (so here go all the element ones). This rules out all the `Document`, `Window`, `DOMTokenList`, `Event`...

`Object.defineProperty` is only used in the polyfills so excluded from the linting (if you know of a way to add them back in the linting for this purpose, well happy to enable it). 

Good spot on `trim` though. It should be have been caught by the linting as [there's a rule for that](https://eslint-community.github.io/eslint-plugin-es-x/rules/no-string-prototype-trim.html) and `accordion.mjs` and `normalise-dataset.mjs` use it. I'll have a look.

Reply:
@romaricpascal Are there alternatives that will catch everything (or more than this)?

Reply:
Ah, reading the docs for `no-string-prototype-trim`, the rule doesn't do anything unless you set the plugin to be `aggressive`.

>This rule is silent by default because it's hard to know types. You need to configure [the aggressive mode](https://eslint-community.github.io/eslint-plugin-es-x/#the-aggressive-mode) or TypeScript in order to enable this rule.

The plugin mentions TypeScript, I'll have a quick try over the `fix-types` branch to see if it does pick it up then (that would be pretty neat).

Reply:
@romaricpascal Had a quick try and the prototype checks came to life 😊 

Turns out we don't need `aggressive: true` with [`@typescript-eslint/parser`](https://www.npmjs.com/package/@typescript-eslint/parser) 🎉 

For comparison: https://github.com/alphagov/govuk-frontend/compare/lint-types...eslint-plugin-es-x-opt-in

```console
/path/to/project/govuk-frontend/src/govuk/components/accordion/accordion.mjs
  145:49  error  ES5 'Function.prototype.bind' method is forbidden  es-x/no-function-prototype-bind
  149:46  error  ES5 'Function.prototype.bind' method is forbidden  es-x/no-function-prototype-bind
  167:5   error  ES5 'Function.prototype.bind' method is forbidden  es-x/no-function-prototype-bind
  178:41  error  ES5 'Function.prototype.bind' method is forbidden  es-x/no-function-prototype-bind
  342:5   error  ES5 'Function.prototype.bind' method is forbidden  es-x/no-function-prototype-bind
  383:25  error  ES5 'String.prototype.trim' method is forbidden    es-x/no-string-prototype-trim
  388:25  error  ES5 'String.prototype.trim' method is forbidden    es-x/no-string-prototype-trim
```